mindmeld/models/nn_utils/input_encoders.py

```diff
@@ -0,0 +1,937 @@
+                  "particular dimension, either pass-in the 'emb_dim' param or provide a " \
+                  "valid 'embedder_type' param."
+            raise ValueError(msg)
+
+    def dump(self, dump_folder):
+        if os.path.isfile(dump_folder):
+            msg = f"Path input to 'dump' method must be a folder, " \
+                  f"not a file ({dump_folder})"
+            raise ValueError(msg)
+        os.makedirs(dump_folder, exist_ok=True)
+
+        if self.embedder_type == "bert":
+            self.pretrained_model_name_or_path = os.path.join(
+                os.path.abspath(dump_folder), f"{self.embedder_type}_resources",
+                os.path.split(self.pretrained_model_name_or_path)[-1]
+            )
+            self._dump_bert_embedder_to_checkpoint(self.pretrained_model_name_or_path)
+        with open(os.path.join(dump_folder, "text_encoder_config.json"), "w") as fp:
+            params = {k: getattr(self, k) for k in self.params_keys}
+            json.dump(params, fp, indent=4)
+            fp.close()
+
+    @classmethod
+    def load(cls, load_folder):
+        if os.path.isfile(load_folder):
+            msg = f"Path input to 'load' method must be a folder, " \
+                  f"not a file ({load_folder})"
+            raise ValueError(msg)
+
+        encoder = cls()
+        with open(os.path.join(load_folder, "text_encoder_config.json"), "r") as fp:
+            params = json.load(fp)
+            setattr(encoder, "params_keys", set(params.keys()))
+            fp.close()
+        encoder.fit(**params, _disable_loading_embedder=True)
+        if encoder.embedder_type == "bert":
+            encoder._load_bert_embedder_from_checkpoint(encoder.pretrained_model_name_or_path)
+
+        return encoder
+
+    @abstractmethod
+    def batch_encode(self, examples, labels=None, **kwargs) -> Dict[str, Any]:
+        msg = f"Subclass {self.name} need to implement this method"
+        raise NotImplementedError(msg)
+
+    def _load_bert_embedder_from_checkpoint(self, pretrained_model_name_or_path):
+        model_bunch = HuggingfaceTransformersContainer(
+            pretrained_model_name_or_path, reload=True).get_model_bunch()
+        self._bert_config = model_bunch.config
+        self._bert_tokenizer = model_bunch.tokenizer
+        self._bert_model = model_bunch.model.to(self.device)
+        if not self.update_embeddings:
+            for param in self._bert_model:
+                param.requires_grad = False
+
+    def _dump_bert_embedder_to_checkpoint(self, ckpt_folder):
+        os.makedirs(ckpt_folder, exist_ok=True)
+        self._bert_config.save_pretrained(ckpt_folder)
+        self._bert_tokenizer.save_pretrained(ckpt_folder)
+        self._bert_model.save_pretrained(ckpt_folder)
+
+
+# sub-classes that define a custom `batch_encode` on top of base encoders
+
+
+class SeqClsEncoderForEmbLayer(WordEmbeddingsBasedEncoder):
```

Comment:
Its not immediately clear why SeqClsEncoderForEmbLayer inherit from WordEmbeddingsBasedEncoder. I thought it should be the other way, since SeqClsEncoderForEmbLayer outputs ints and WordEmbeddingsBasedEncoder converts those ints to vectors